### ISSUE TITLE
Overhaul CoreTask and LHCbTask

### ISSUE THREAD

**Issue by mesmith75** [ganga-devs/ganga#1761]
We have discussed this at some point but it seems as if LHCbTask is gaining some traction in the collaboration.

The current implementation is pretty opaque. It would be good to overhaul this and try and make it a little bit more usable.

**Comment by alexanderrichards** [comment]
yes agreed, the entire tasks framework needs properly integrating into the usual Jobs framework

**Comment by CasaisA** [comment]
I'm an extensive user of LHCb tasks and would be happy to give a helping hand, wherever it could be useful.

**Comment by egede** [comment]
> I'm an extensive user of LHCb tasks and would be happy to give a helping hand, wherever it could be useful.

This would be most welcome. The first activity would be some design work - and getting input from somebody who is actually an extensive user is ideal. Maybe first step is to describe your usage, what annoys you in the present solution and where you see missing features. We can then move onto how we would like the API to look like.

**Comment by CasaisA** [comment]
Well, first of all, I use them "extensively" but for a simple purpose: private MC productions. So I mainly use the same features over and over... 

What I usually do is design my task using an official MC model so it has a transform for each step in the simulation model:

* Gauss
* Boole
* Moore L0
* More Hlt1
* Moore Hlt2
* Brunel

In each step the outputfiles are passed as input to the next step by using `TaskChainInput`.

An example of the type of tasks that I use may be found [here](https://gitlab.cern.ch/acasaisv/diphotonanalysis/-/blob/master/simulation/ALPs/ganga-withgridpack/ALPs-ganga.py).

My main concerns are the following:

* Tasks usually get stuck for no apparent reason
     * Some transform finishes successfully but the next step won't be launched ever
* Tasks cannot be successfully forced to continue when some jobs repeatedly fail
     * Some times the majority of the subjobs have finished, with the exception of a small fraction. Even if the job is forced to completed status, the majority of the times the task won't launch the next transform.
* `TaskChainInput` concentrates the majority of the events in the first output DST of the last step. 

Thanks!

**Comment by ryuwd** [comment]
Apologies to add to an old thread - 

I'm not an extensive user of Tasks or LHCbTasks, but I've been using `LHCbTask` recently at my home institute (Edinburgh). One thing I noticed is that `LHCbTask` can be a little difficult to run on other backends that aren't `Dirac`, which probably discourages people from trying to run their tasks at their home institutes.

For example, `delete_chain_input` assumes a `Dirac()` backend since it checks a `backend.status` attribute which e.g. `SGE` does not have. Also, on Unit creation the splitter is set by default to `SplitByFiles`, which also depends directly on `Dirac` / `DiracFile` and doesn't work with `MassStorageFile` (I opened an issue #1852).

I got around these issues by setting my `splitter = None` if the backend was not `Dirac` and leaving `delete_chain_input = False` but the former required source code adjustments in my local install of ganga. (this is now addressed in #1848 #1851)
